pkg/cluster/clustermsi.go

```diff
@@ -62,40 +54,12 @@ func (m *manager) ensureClusterMsiCertificate(ctx context.Context) error {
 		return err
 	}
 
-	now := time.Now()
-
-	var expirationDate time.Time
-	if m.env.FeatureIsSet(env.FeatureUseMockMsiRp) {
-		expirationDate = now.AddDate(0, 0, mockMsiCertValidityDays)
-	} else {
-		identity, err := getSingleExplicitIdentity(msiCredObj)
-		if err != nil {
-			return err
-		}
-		if identity.NotAfter == nil {
-			return errors.New("unable to pull NotAfter from the MSI CredentialsObject")
-		}
-
-		// The OpenAPI spec for the MI RP specifies that NotAfter will be "in the format 2017-03-01T14:11:00Z".
-		expirationDate, err = time.Parse(time.RFC3339, *identity.NotAfter)
-		if err != nil {
-			return err
-		}
-	}
-
-	raw, err := json.Marshal(msiCredObj)
+	_, parameters, err := dataplane.FormatManagedIdentityCredentialsForStorage(m.doc.ID, *msiCredObj)
```

Comment:
TODO for me - need another PR to expose the name (the `_` output of this func) for cases where we need to look up the name based on the ID, without the rest of the msi credentials.

Reply:
Fixed in https://github.com/Azure/msi-dataplane/pull/45